lib/cookie.js

```diff
@@ -1332,7 +1354,9 @@ class CookieJar {
     if (
       secure == null &&
       context.protocol &&
-      (context.protocol == "https:" || context.protocol == "wss:")
+      (context.protocol == "https:" ||
+        context.protocol == "wss:" ||
+        isSpecialDomain(host))
```

Comment:
Let's also leave a comment with reference to the RFC about why this is check is being made.

Reply:
Done.